include/alpaka/api/syclGeneric/Device.hpp

```diff
@@ -106,8 +106,10 @@ namespace alpaka::onHost
                     sycl::aligned_alloc_device(alignment, memSizeInByte, sycl_device, sycl_context));
                 auto deleter = [ctx = sycl_context, ptr]()
                 {
-                    void* ptrToFree
-                        = reinterpret_cast<void*>(const_cast<std::remove_volatile_t<ALPAKA_TYPEOF(ptr)>>(ptr));
+                    void* ptrToFree = reinterpret_cast<void*>(
+                        const_cast<
```

Comment:
could you introduce in `util.hpp`

```C++
template<typename T>
using RemoveVolatileFromPointee_t = std::add_pointer_t<std::remove_volatile_t<std::remove_pointer_t<T>>>;
```

But the other question is, we cast the pointer anyway to `void*` why do we need to remove the volatile before?